tests/src/test/java/alluxio/client/cli/job/DistributedCmdTest.java

```diff
@@ -0,0 +1,145 @@
+    mRetryPolicy = new CountingRetry(5);
+
+    mLoadURIStatus = Mockito.mock(URIStatus.class);
+    mMigrateURIStatus = Mockito.mock(URIStatus.class);
+    mPersistURIStatus = Mockito.mock(URIStatus.class);
+    mNoopPlanURIStatus = Mockito.mock(URIStatus.class);
+  }
+
+  private void initialize() throws Exception {
+    Mockito.when(mLoadURIStatus.getLength()).thenReturn(LOAD_FILE_LENGTH);
+
+    Mockito.when(mLoadJobConfig.getFilePath()).thenReturn(LOAD);
+    Mockito.when(mMigrateJobConfig.getSource()).thenReturn(MIGRATE);
+    Mockito.when(mPersistJobConfig.getFilePath()).thenReturn(PERSIST);
+
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(LOAD))).thenReturn(mLoadURIStatus);
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(LOAD)).getLength())
+            .thenReturn(LOAD_FILE_LENGTH);
+
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(MIGRATE))).thenReturn(mMigrateURIStatus);
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(MIGRATE)).getLength())
+            .thenReturn(MIGRATE_FILE_LENGTH);
+
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(PERSIST))).thenReturn(mPersistURIStatus);
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(PERSIST)).getLength())
+            .thenReturn(PERSIST_FILE_LENGTH);
+
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(NOOP_PLAN))).thenReturn(mNoopPlanURIStatus);
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(NOOP_PLAN)).getLength())
+            .thenReturn(NOOP_PLAN_FILE_LENGTH);
+  }
+
+  @Test
+  public void testSupportedConfigForCompleteStatus() throws Exception {
+    initialize();
+    //Run for supported configs.
+    DistributedCmdMetrics.incrementForCompleteStatusWithRetry(
+            mLoadJobConfig, mFileSystem, mRetryPolicy);
+    DistributedCmdMetrics.incrementForCompleteStatusWithRetry(
+            mMigrateJobConfig, mFileSystem, mRetryPolicy);
+    DistributedCmdMetrics.incrementForCompleteStatusWithRetry(
+            mPersistJobConfig, mFileSystem, mRetryPolicy);
+
+    ////should expect function calls to fileSystem.getStatus(new AlluxioURI(filePath)).getLength(),
+    // and cooresponding file sizes obtained.
+    double loadFileSize = MetricsSystem.getMetricValue(
+            MetricKey.MASTER_JOB_DISTRIBUTED_LOAD_FILE_SIZE.getName()).getValue();
+    double migrateFileSize = MetricsSystem.getMetricValue(
+            MetricKey.MASTER_MIGRATE_JOB_FILE_SIZE.getName()).getValue();
+    double persistFileSize = MetricsSystem.getMetricValue(
+            MetricKey.MASTER_ASYNC_PERSIST_FILE_SIZE.getName()).getValue();
+
+    assertEquals(loadFileSize, LOAD_FILE_LENGTH, 0);
+    assertEquals(migrateFileSize, MIGRATE_FILE_LENGTH, 0);
+    assertEquals(persistFileSize, PERSIST_FILE_LENGTH, 0);
+  }
+
+  @Test
+  public void testUnSupportedConfigForCompleteStatus() throws Exception {
+    MetricsSystem.resetAllMetrics();
+    //Run for an unsupported config NoopPlanConfig.
+    DistributedCmdMetrics.incrementForCompleteStatusWithRetry(
```

Comment:
rather than picking a single example class that is not supported, is it possible to use something like a ServiceLoader to get all the possible implementations of `PlanConfig` so that the unit test has coverage for all types and does not need to be updated as new types get added?

Reply:
Good suggestion, but currently we don't have this kind of config loader to load all types of PlanConfig. Will make a note, we can add this feature in the future.